frontend/src/app/projects/models/model-source/t4c/manage-t4c-model/manage-t4c-model.component.ts

```diff
@@ -0,0 +1,200 @@
+
+  fetchT4CInstances(): Observable<T4CInstance[]> {
+    return this.t4cInstanceService.listInstances().pipe(
+      tap((instances) => {
+        this.t4cInstances = instances;
+        this.form.controls.t4cInstanceId.enable({ emitEvent: false });
+        this.form.controls.t4cInstanceId.patchValue(
+          this.form.controls.t4cInstanceId.value
+        );
+      })
+    );
+  }
+
+  updateRepositoryListOnInstanceChange(): void {
+    this.form.controls.t4cInstanceId.valueChanges.subscribe(
+      (t4cInstanceId: number | null) => {
+        if (t4cInstanceId) {
+          this.t4cRepositoryService
+            .getT4CRepositories(t4cInstanceId)
+            .subscribe((repositories) => {
+              this.t4cRepositories = repositories;
+              this.form.controls.t4cRepositoryId.enable({
+                emitEvent: false,
+              });
+            });
+        }
+      }
+    );
+  }
+
+  updateProjectListOnRepositoryChange(): void {
+    this.form.controls.t4cRepositoryId.valueChanges.subscribe(() => {
+      this.form.controls.name.enable({ emitEvent: false });
+    });
+  }
+
+  patchFormWithExistingT4CModel(): Observable<T4CModel | undefined> {
+    return this.t4cModelService._t4cModel.pipe(
+      tap((model: T4CModel | undefined) => {
+        if (model) {
+          this.editing = true;
+          this.t4cModel = model;
+
+          this.form.controls.t4cInstanceId.setValue(
+            model.repository.instance.id,
+            { emitEvent: true }
+          );
+
+          this.form.controls.t4cRepositoryId.patchValue(model.repository.id);
+          this.form.controls.name.patchValue(model.name);
+          this.form.enable({ emitEvent: false });
+        }
+      })
+    );
+  }
+
+  onSubmit() {
+    if (this.form.valid) {
+      if (this.t4cModelService.t4cModel) {
+        this.t4cModelService
+          .patchT4CModel(
+            this.projectService.project!.slug,
+            this.modelService.model!.slug,
+            this.t4cModelService.t4cModel.id,
+            this.form.value as SubmitT4CModel
+          )
+          .subscribe((model) => {
+            this.t4cModelService._t4cModel.next(model);
+            this.patchFormWithExistingT4CModel();
+
+            this.toastService.showSuccess(
+              'TeamForCapella successfully updated',
+              ''
+            );
+          });
+      } else {
+        this.t4cModelService
+          .createT4CModel(
+            this.projectService.project!.slug,
+            this.modelService.model!.slug,
+            this.form.value as SubmitT4CModel
+          )
+          .subscribe((_) => {
+            if (this.asStepper) {
+              this.create.emit(true);
+            } else {
+              this.router.navigate(['../..'], { relativeTo: this.route });
+            }
+
+            this.toastService.showSuccess(
+              'TeamForCapella successfully created',
+              ''
+            );
+          });
+      }
+    }
+  }
```

Comment:
```suggestion
  onSubmit() {
    if (this.form.invalid) {
      return;
    }
    if (this.t4cModelService.t4cModel) {
      this.t4cModelService
        .patchT4CModel(
          this.projectService.project!.slug,
          this.modelService.model!.slug,
          this.t4cModelService.t4cModel.id,
          this.form.value as SubmitT4CModel
        )
        .subscribe((model) => {
          this.t4cModelService._t4cModel.next(model);
          this.patchFormWithExistingT4CModel();

          this.toastService.showSuccess(
            'TeamForCapella successfully updated',
            ''
          );
        });
    } else {
      this.t4cModelService
        .createT4CModel(
          this.projectService.project!.slug,
          this.modelService.model!.slug,
          this.form.value as SubmitT4CModel
        )
        .subscribe((_) => {
          if (this.asStepper) {
            this.create.emit(true);
          } else {
            this.router.navigate(['../..'], { relativeTo: this.route });
          }

          this.toastService.showSuccess(
            'TeamForCapella successfully created',
            ''
          );
        });
    }
  }
```
This reduces the nesting level here, which makes it a little easier to read.

Reply:
I like. Applied.